pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -71,3 +85,84 @@ func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstora
 func (p RESTStorageProvider) GroupName() string {
 	return flowcontrol.GroupName
 }
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	return PostStartHookName, func(hookContext genericapiserver.PostStartHookContext) error {
+		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		// ensure the pre-defined objects if the exempt flow-schema is present.
+		shouldEnsure, err := shouldEnsureAllPredefined(flowcontrolClientSet)
+		if err != nil {
+			return err
+		}
+		if shouldEnsure {
+			return ensureAllPredefined(
+				flowcontrolClientSet,
+				flowcontrolbootstrap.PredefinedFlowSchemas(),
+				flowcontrolbootstrap.PredefinedPriorityLevelConfigurations(),
+			)
+		}
+		return nil
+	}, nil
+}
+
+func shouldEnsureAllPredefined(flowcontrolClientSet flowcontrolclient.FlowcontrolV1alpha1Interface) (bool, error) {
```

Comment:
It looks to me like this function is returning the opposite of what its name says.

Reply:
updated, and added godoc to clarity the logic